lib/screens/wods/components/delete_wod_button.dart

```diff
@@ -0,0 +1,64 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+//My imports
+import 'package:kabod_app/core/presentation/constants.dart';
+import 'package:kabod_app/screens/home/repository/wod_repository.dart';
+import 'package:kabod_app/screens/wods/add_wod.dart';
+
+class DeleteWodButton extends StatelessWidget {
+  const DeleteWodButton({
+    Key key,
+    @required this.widget,
+  }) : super(key: key);
+
+  final AddWodScreen widget;
```

Comment:
You should never ever do this. You are saving a widget in a variable just to use an attribute of it (part of its state)... Widgets are not supposed to be used to save and get state. Ever.

Here you should maybe create a model (like WodEditorModel) by using `ChangeNotifier` to save the currentWod state and use it though the required screens.